### PR TITLE
feat: StarSearch as a standalone component

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

This is a refactor that pulls out the StarSearch chat from the StarSearch page as a precursor to the work required for #3508.

The UI looks and behaves the same as it previously did in #3324. I've also added tests to ensure the URLs for prompt sharing and copy the shared prompt URL to clipboard functionality.

I'm not refactoring some things at the moment:

* Suggested prompts - They're staying in the StarSearchChat component for now, but once this is used in Workspaces, we may want to provide suggested prompts for the repositories and users in their workspace.
* The endpoint will be different for streaming responses in the context of workspaces.

## Related Tickets & Documents

Relates to #3508

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/833231/eb481748-ca49-4f76-9bb3-e9c372c7d3d4

## Steps to QA

**Share a Prompt**

1. Log in to StarSearch
2. Choose a canned prompt or write your own.
3. When the response is finished, notice the share menu at the bottom right of the last response.
4. Click on it to show share to Twitter, LinkedIn and copy link.
5. Try those out.
7. Copy the shared prompt link

**Try the shared prompt**

**Logged In User**

1. Paste the shared prompt link in the browser from the steps before.
2. The prompt and response load even though you're logged out.
3. Try another prompt once the response is completed.
4. Nologin modal appears and the streaming response comes in.

**Logged Out User**

1. Paste the shared prompt link in the browser from the steps before.
2. The prompt and response load even though you're logged out.
3. Try another prompt once the response is completed.
4. A login modal appears and you are forced to log in to continue prompting.
5. 
## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

## [optional] What gif best describes this PR or how it makes you feel?

Starsearch

### CODE REVIEW

File: e2e/star-search.spec.ts [hunk 1]
```diff
@@ -139,19 +141,45 @@ test("StarSearch shared prompt (Logged Out Flow)", async ({ page }) => {
...
 
-  await expect(page.getByRole("menuitem", { name: "Share to Twitter/X", exact: true })).toBeVisible();
-  await expect(page.getByRole("menuitem", { name: "Share to LinkedIn", exact: true })).toBeVisible();
-  await expect(page.getByRole("menuitem", { name: "Copy link", exact: true })).toBeVisible();
+  const shareToTwitterMenuItem = page.getByRole("menuitem", { name: "Share to Twitter/X", exact: true });
+  const shareToLinkedInMenuItem = page.getByRole("menuitem", { name: "Share to LinkedIn", exact: true });
+  const copyLinkMenuItem = page.getByRole("menuitem", { name: "Copy link", exact: true });
+
+  await expect(shareToTwitterMenuItem).toBeVisible();
+  await expect(shareToTwitterMenuItem.locator("a")).toHaveAttribute(
```
Comment (nickytonline) [inline e2e/star-search.spec.ts:149]: Added some tests to ensure URLs are correct.

File: e2e/star-search.spec.ts [hunk 2]
```diff
@@ -139,19 +141,45 @@ test("StarSearch shared prompt (Logged Out Flow)", async ({ page }) => {
...
+  await expect(shareToLinkedInMenuItem).toBeVisible();
+  await expect(shareToLinkedInMenuItem.locator("a")).toHaveAttribute(
+    "href",
+    `https://www.linkedin.com/sharing/share-offsite/?url=${encodeURIComponent(
+      BASE_URL
+    )}%2Fstar-search%3Fprompt%3DWho%2Bare%2Bthe%2Bmost%2Bprevalent%2Bcontributors%2Bto%2Bthe%2BTypeScript%2Becosystem%253F`
+  );
+  await expect(copyLinkMenuItem).toBeVisible();
+
+  if (["webkit", "Desktop Safari", "Mobile Safari"].includes(browserName)) {
```
Comment (nickytonline) [inline e2e/star-search.spec.ts:165]: Added a test to ensure copying a link to the clipboard works. Currently not supported by webkit/Safari. See https://github.com/microsoft/playwright/issues/13037

File: playwright.config.ts [hunk 3]
```diff
@@ -34,12 +34,20 @@ export default defineConfig({
   projects: [
     {
       name: "chromium",
-      use: { ...devices["Desktop Chrome"] },
+      use: { ...devices["Desktop Chrome"], permissions: ["clipboard-read", "clipboard-write"] },
```
Comment (nickytonline) [inline playwright.config.ts:37]: I added these settings in the Playwright configuration to allow access to the clipboard. Thanks to this article, https://playwrightsolutions.com/how-do-i-access-the-browser-clipboard-with-playwright/ and this GitHub comment, https://github.com/microsoft/playwright/issues/13037#issuecomment-1667307993.

**Comment by BekahHW** [comment]
@nickytonline will the url shortener work once it's not in deploy preview?

image

**Comment by nickytonline** [comment]
> @nickytonline will the url shortener work once it's not in deploy preview?
> 
> 
> 
> image

@BekahHW , yes they will. For the deploy previews we use the free version and its cache gets full. See this comment from my previous PR.

**Review by BekahHW — APPROVED** [review]

**Review by jpmcb — APPROVED** [review]
Took it for a spin, looks great! 🚀

**Pull request merged**